src/Spectre.Data/RoiIo/IRoiUtilites.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * IRoiUtilities.cs
+ * Interface for RoiUtilities.
+
+   Copyright 2017 Roman Lisak
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Spectre.Data.Datasets;
+
+namespace Spectre.Data.RoiIo
+{
+    /// <summary>
+    /// Interface for RoiUtilities.
+    /// </summary>
+    public interface IRoiUtilites
```

Comment:
This interface gathers three different functionalities, that should be separated.

I see this as two-layer abstraction:

1. Raw I/O - similar functionality to what we have here, decomposed into two classes:
  a. `RoiReader` - created for a *dataset* (to be precise, created for a directory path). Provides discovery (as `ListRoisFromDirectory`) and reading a ROI of a specified name.
  b. `RoiWriter` - created the same way the `RoiReader` is. Allows to write ROI under specified name.
2. A kind of repository for ROIs. It would be created the same way the above raw I\O is. How should it work from the outside? Similarly to `Dictionary<string, Roi>`: we should be able to list all the ROIs by name, read by name and store by name (with interface similar to usual C# container).

Please investigate, how it can be done in a line with .NET conventions (to provide similar interface than `Dictionary` does). It will be easy to use, while providing the same functionality.